packages/builder/src/components/automation/SetupPanel/layouts/APIRequest.svelte

```diff
@@ -0,0 +1,311 @@
+  import { queries, datasources } from "@/stores/builder"
+  import ApiParamSelector from "./APIParamSelector.svelte"
+  import CodeEditor from "@/components/common/CodeEditor/CodeEditor.svelte"
+  import { runtimeToReadableBinding, getAuthBindings } from "@/dataBinding"
+  import { EditorModes } from "@/components/common/CodeEditor"
+  import KeyValueBuilder from "@/components/integration/KeyValueBuilder.svelte"
+  import { configFromIntegration } from "@/stores/selectors"
+  import { IntegrationTypes } from "@/constants/backend"
+  import { goto, params } from "@roxi/routify"
+
+  export let bindings: EnrichedBinding[] | undefined = undefined
+  export let block: AutomationStep | undefined = undefined
+  export let context: AutomationContext | undefined
+
+  let value: any
+  let authBindings: any[] = getAuthBindings()
+  let selectedDatasourceId: string | undefined
+
+  // Include any custom bindings from the REST API data section UI to ensure completeness
+  // when parsing to readable
+  $: restBindings = [...(bindings || []), ...authBindings]
+
+  // The step input properties
+  $: inputData = automationStore.actions.getInputData(block)
+  $: fieldKey = "query"
+  $: {
+    value = getInputValue(inputData, fieldKey)
+  }
+  $: if (value) {
+    selectedDatasourceId = undefined
+  }
+
+  // Include only REST API sources
+  $: restSources = $datasources?.list?.filter(d => d.source === SourceName.REST)
+
+  $: targetSource = selectedDatasourceId || query?.datasourceId
+
+  // Source for current query, if any
+  $: dataSource =
+    restSources?.find(ds => ds._id === targetSource) || restSources?.[0]
+
+  $: restQueries = $queries.list.filter(q => {
+    return q.datasourceId === dataSource._id
+  })
+
+  // The configured query
+  $: query = $queries.list.find(query => query._id === value?.queryId)
+
+  // QUERY DETAILS
+
+  $: authId = query?.fields?.authConfigId
+  $: auth = (dataSource?.config?.authConfigs || []).find(
+    (auth: any) => auth._id === authId
+  )
+
+  // Parse the request for display
+  $: requestPreview = getRESTPreview(query)
+  $: readablePreview = runtimeToReadableBinding(restBindings, requestPreview)
+
+  // Parse the bindings of the request body when present.
+  $: readableBody =
+    query?.fields?.bodyType !== BodyType.NONE
+      ? runtimeToReadableBinding(restBindings, query?.fields?.requestBody)
+      : null
+
+  // Show only enabled headers and make any bindings readable
+  $: queryHeaders = query?.fields?.headers
+  $: parsedHeaderEntries = Object.entries(queryHeaders || {})
+    .filter(([key]: [string, string]) => {
+      return !query?.fields?.disabledHeaders?.[key]
+    })
+    .map(([key, val]: [string, string]) => {
+      return [key, runtimeToReadableBinding(restBindings, val)]
+    })
+  $: parsedHeaders = Object.fromEntries(parsedHeaderEntries)
+
+  const getRESTPreview = (query: Query | undefined) => {
+    if (!query) return
+    return query?.fields?.queryString
```

Comment:
NIT: extracting the `{queryString, path} = query.fields` might make this more readable